rotkehlchen/db/upgrades/v45_v46.py

```diff
@@ -63,4 +74,58 @@ def _move_extra_data(write_cursor: 'DBCursor') -> None:
         )
         write_cursor.execute('ALTER TABLE evm_events_info DROP COLUMN extra_data;')
 
+    @progress_step(description='Converting asset movements to history events')
+    def move_asset_movements(write_cursor: 'DBCursor') -> None:
+        new_events: list[AssetMovement] = []
+        write_cursor.execute(
+            'SELECT id, location, category, address, transaction_id, timestamp, asset, '
+            'amount, fee_asset, fee, link FROM asset_movements',
+        )
+
+        for row in write_cursor:
+            try:
+                new_events.extend(create_asset_movement_with_fee(
+                    timestamp=ts_sec_to_ms(row[5]),
+                    location=Location.deserialize_from_db(row[1]),
+                    event_type=HistoryEventType.DEPOSIT if row[2] == 'A' else HistoryEventType.WITHDRAWAL,  # noqa: E501
+                    asset=Asset(row[6]),
+                    amount=FVal(row[7]),
+                    fee_asset=Asset(row[8]),
+                    fee=FVal(row[9]),
+                    extra_data=maybe_set_transaction_extra_data(
+                        address=row[3],
+                        transaction_id=row[4],
+                        extra_data={'url': row[10]} if row[10] is not None and row[10] not in ('', 'None') else None,  # in coinbase we were storing 'None' as string.  # noqa: E501
+                    ),
+                ))
+            except DeserializationError as e:
+                log.error(f'Failed to deserialize row {row} due to {e}. Skipping.')
+
+        binding_tuples = ((
+            event.entry_type.value,
+            event.event_identifier,
+            event.sequence_index,
+            int(event.timestamp),
+            event.location.serialize_for_db(),
+            event.location_label,
+            event.asset.identifier,
+            str(event.balance.amount),
+            str(event.balance.usd_value),
+            event.notes,
+            event.event_type.serialize(),
+            event.event_subtype.serialize(),
+            json.dumps(event.extra_data) if event.extra_data else None,
+        ) for event in new_events)
```

Comment:
Wouldn't `event.serialize_for_db()` work here?

Reply:
yes but I did it like this in case we change the serialization logic due to changes in the schema

Reply:
ok, makes sense.